File: JavaToKotlin/app/src/main/java/io/dustin/apps/common/utils/CommonUtil.java
package io.dustin.apps.common.utils;

import io.dustin.apps.common.provider.ApplicationContextProvider;
import org.springframework.context.ApplicationContext;
import org.springframework.stereotype.Component;

@Component
public class CommonUtil {

    public static <T> T getBean(String beanId, Class<T> clazz) {
        ApplicationContext applicationContext = ApplicationContextProvider.getApplicationContext();
        if(applicationContext == null) {
            throw new NullPointerException("Not Initialize Spring Container!");
        }
        return (T)applicationContext.getBean(beanId);
    }

}

File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/common/utils/CommonUtil.kt
package io.dustin.apps.common.utils

import io.dustin.apps.common.provider.ApplicationContextProvider

@Component
object CommonUtil {
    fun <T> getBean(beanId: String?, clazz: Class<T?>?): T? {
        val applicationContext: ApplicationContext = ApplicationContextProvider.getApplicationContext()
            ?: throw NullPointerException("Not Initialize Spring Container!")
        return applicationContext.getBean(beanId) as T
    }
}
